src/renderer/src/hooks/useFetchModels.ts

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+
+interface OpenRouterModel {
+  id: string;
+  // others not needed
+}
+
+interface OpenRouterResponse {
+  data: OpenRouterModel[];
+}
+
+export const useFetchModels = (apiKey: string, url: string) => {
```

Comment:
I would rather go for 2 separate hooks here that are specific for both the providers - `useRequestyModels` and `useOpenRouterModels` and have the URL as the internal implementation detail. This way they can be easily reused in other parts if needed.

It's OK that they will have most of the code same, but having it separated makes it also more understandable for other developers.